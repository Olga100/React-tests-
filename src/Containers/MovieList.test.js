import React from 'react';
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {shallow} from 'enzyme';
import {MovieListView} from './MovieList';


describe('<MovieListView/>', () => {

    it('Renders  component with list of the movie', () => {

        const movies =
            [
                {
                    id: 1,
                    title: "title",
                    posterUrl: "http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg",
                    genres: ["horror"],
                    actors: ["Vasya", "Sasha"],
                    director: "Petya",
                    description: "This is a very good movie!",
                    stars: 4,
                    likes: 42
                },
                {
                    id: 2,
                    title: "title11",
                    posterUrl: "http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg",
                    genres: ["romcom"],
                    actors: ["Vasya", "Sasha"],
                    director: "Petya",
                    description: "This is a very good movie too!",
                    stars: 5,
                    likes: 43
                }
            ];

        const mockStore = configureMockStore();
        const store = mockStore({});

        const component = renderer.create(
            <Provider store={store}>
                <MovieListView movies={movies}/>
            </Provider>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});
