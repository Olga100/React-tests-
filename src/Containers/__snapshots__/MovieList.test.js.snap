// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieListView/> Renders  component with list of the movie 1`] = `
<div
  className="leftPart"
>
  <div
    className="FieldSortSearch"
  >
    <p>
      Sort movies
    </p>
    <div>
      <button
        className="sortByLikies"
        onClick={undefined}
        type="button"
      >
        <small>
           by likies
        </small>
      </button>
      <button
        className="sortByRating"
        onClick={undefined}
        type="button"
      >
        <small>
           by rating
        </small>
      </button>
    </div>
    <div
      className="fieldInput"
    >
      <button
        className="submit"
        onClick={[Function]}
        type="submit"
      >
        ðŸ”Ž
      </button>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search by name"
        type="search"
        value=""
      />
    </div>
  </div>
  <div
    className="movie-thumbnail-title"
    onClick={[Function]}
  >
    <div
      className="title"
    >
      <small>
        title
      </small>
    </div>
    <div
      className="likes"
    >
      <button
        className="thumb"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-o-up"
        />
      </button>
      <button
        className="thumb"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-o-down"
        />
      </button>
      <small>
        likes
      </small>
      <hr />
      <span
        className="amountLike"
      >
        42
      </span>
    </div>
    <img
      alt="poster"
      className="image"
      src="http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg"
    />
    <div
      className="clear"
    />
    <div
      className="stars"
    >
      <span
        style={
          Object {
            "cursor": "default",
          }
        }
      >
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜†
        </span>
      </span>
    </div>
  </div>
  <div
    className="movie-thumbnail-title"
    onClick={[Function]}
  >
    <div
      className="title"
    >
      <small>
        title11
      </small>
    </div>
    <div
      className="likes"
    >
      <button
        className="thumb"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-o-up"
        />
      </button>
      <button
        className="thumb"
        onClick={[Function]}
        type="button"
      >
        <i
          aria-hidden="true"
          className="fa fa-thumbs-o-down"
        />
      </button>
      <small>
        likes
      </small>
      <hr />
      <span
        className="amountLike"
      >
        43
      </span>
    </div>
    <img
      alt="poster"
      className="image"
      src="http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg"
    />
    <div
      className="clear"
    />
    <div
      className="stars"
    >
      <span
        style={
          Object {
            "cursor": "default",
          }
        }
      >
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
        <span
          className="star"
          onClick={[Function]}
        >
          â˜…
        </span>
      </span>
    </div>
  </div>
</div>
`;
